File: 프로그래머스/0/120895. 인덱스 바꾸기/인덱스 바꾸기.java
class Solution {
    public String solution(String my_string, int num1, int num2) {
        String answer = "";
        String[] strArr = my_string.split("");
        
        for(int i = 0; i < strArr.length; i++) {
            
            if(i == num1) {
                answer += strArr[num2];    
                continue;
            } else if(i == num2) {
                answer += strArr[num1];    
                continue;
            }
            
            answer += strArr[i];
        }
        
        return answer;
    }
}

// 다른 풀이
String answer = "";

char[] ch = my_string.toCharArray();

ch[num1] = my_string.charAt(num2);
ch[num2] = my_string.charAt(num1);

answer = String.valueOf(ch);
return answer;

// 다른 풀이
List<String> list = Arrays.stream(myString.split(""))
                          .collect(Collectors.toList());

Collections.swap(list, num1, num2);
return String.join("", list);

File: 프로그래머스/0/120895. 인덱스 바꾸기/인덱스 바꾸기.py
def solution(my_string, num1, num2):
    answer = ''
    idx = 0
    
    for str in my_string:
        if idx == num1:
            answer += my_string[num2]
            idx += 1
        elif idx == num2:
            answer += my_string[num1]
            idx += 1
        else:
            answer += str
            idx += 1
    
    return answer